### DISCUSSION TITLE
Support for reverse connect

### DISCUSSION THREAD

**Discussion by cpipero** [FreeOpcUa/opcua-asyncio#570] [Q&A]
I could not find any information on Reverse Connect (either on existing features or future roadmap) and was wondering if there is any plan to support.

See OPC UA Specs at https://reference.opcfoundation.org/v104/Core/docs/Part7/6.6.5/ .

I'm planning to implement a client but the customer is mandating reverse connect to their servers for firewall constraints.

Thanks for any information about this feature.

**Comment by pmohan6** [comment]
Same question as the author. We have a requirement for reverse connect as well. Is this something that is being worked on or is on the roadmap?

Would really appreciate any information!

**Comment by swamper123** [comment]
**Current Status:** There isn't any possibility atm to support any Profiles or Facettes. Neither server not client. Only a solid "sum of features" are supported, which leads to a still functioning server. (correct me if I am wrong @oroulet, perhaps my expression is a bit missleading). I listed all needed Parts for a `Nano Embedded Device 2017 Server` [here](https://github.com/FreeOpcUa/opcua-asyncio/projects/1).


What I (and I bet some others as well) wish for is to support different kind of Profiles and Facets in general (Server Profiles prefered).
